### PR TITLE
error state component

### PR DESCRIPTION
**Pull request by ShaiahWren**
Hi, 

This pr takes the place of https://github.com/ansible/ansible-ui/pull/280, which is now closed. The UI should show no changes.  

Let me know if you have any questions or concerns. 

Thank you!

### CODE REVIEW

File: framework/components/EmptyStateError.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import {
+  Title,
+  EmptyStateBody,
+  EmptyState,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  Button,
+} from '@patternfly/react-core';
+
+export function EmptyStateError(props: { message?: string }) {
```
Comment (ZitaNemeckova) [inline framework/components/EmptyStateError.tsx:11]: The default messages needs to be translated.

```
import { useTranslation } from 'react-i18next';
...
const { t } = useTranslation();
const title = t('Something went wrong');
...
```

Comment (ZitaNemeckova) [reply framework/components/EmptyStateError.tsx:12]: Yep :)

File: framework/PageTable/PageTable.tsx [hunk 2]
```diff
@@ -164,43 +165,29 @@ export function PageTable<T extends object>(props: PageTableProps<T>) {
...
-            color="var(--pf-global--danger-color--100)"
-          />
-          <Title headingLevel="h2" size="lg">
-            {/* Unable to connect */}
-            {props.errorStateTitle}
-          </Title>
-          {/* <EmptyStateBody>There was an error retrieving data. Check your connection and reload the page.</EmptyStateBody> */}
-          <EmptyStateBody>{error.message}</EmptyStateBody>
-        </EmptyState>
+        <EmptyStateError message={error.message} />
```
Comment (ZitaNemeckova) [inline framework/PageTable/PageTable.tsx:168]: @ShaiahWren I think it makes sense to allow `title` to be an optional param as well. WDYT?

Comment (ShaiahWren) [reply framework/PageTable/PageTable.tsx:168]: Sure, I can do that.

Comment (ShaiahWren) [reply framework/PageTable/PageTable.tsx:168]: Ah, so title is already an optional param called 'message' - in `EmptyStateError` the default title is `const title = t('Something went wrong');`.  Perhaps `error.message` is not the correct alternative to the default message. I'm trying to reconcile what is suggested in the mockups with what we already have:
Screenshot 2023-03-29 at 9 12 49 AM

**Review by ZitaNemeckova — APPROVED** [review]
LGTM 👍 needs rebase

**Review by vidyanambiar — COMMENTED** [review]
- I see some styling changes in the Empty state introduced by this PR. The content width is smaller and introduces more wrapped text in the empty state messages. Is this an intended styling change?
- 
image
image

- In the error scenario it might be good to have more vertical spacing between the error message and the prompt "Please try refreshing.." to be consistent with the spacing above the error message:
- 
image

File: framework/PageTable/PageTable.tsx [hunk 3]
```diff
@@ -6,13 +6,12 @@ import {
   EmptyStateBody,
   EmptyStateIcon,
   EmptyStateSecondaryActions,
-  EmptyStateVariant,
```
[the remark this answers is not in the document]
Comment (ShaiahWren) [reply framework/PageTable/PageTable.tsx]: HI @vidyanambiar, thank you for your review! My recent changes should address these concerns. Let me know if there are any more questions.

**Comment by vidyanambiar** [comment]
Thanks for addressing the comments, @ShaiahWren. I tested the empty state and it looks good to me. 👍 

I noticed a styling issue in the error state. The "Please refresh.." text shows up in white and is hard to read. 
image

**Review by vidyanambiar — APPROVED** [review]
lgtm

File: framework/PageTable/PageTable.tsx [hunk 4]
```diff
@@ -281,50 +283,55 @@ export function PageTable<T extends object>(props: PageTableProps<T>) {
 
   if (error) {
     return (
-      <EmptyStateDiv>
```
Comment (ZitaNemeckova) [inline framework/PageTable/PageTable.tsx:284]: If `<EmptyStateDiv>` uses are removed everywhere it's definition should be too.

**Review by ZitaNemeckova — APPROVED** [review]
LGTM. Great work!

**Pull request merged**